src/test/java/seedu/address/model/person/UniqueAssignmentListTest.java

```diff
@@ -0,0 +1,173 @@
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalAssignments.ALICE;
+import static seedu.address.testutil.TypicalAssignments.BOB;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.person.exceptions.AssignmentNotFoundException;
+import seedu.address.model.person.exceptions.DuplicateAssignmentException;
+import seedu.address.testutil.AssignmentBuilder;
+
+public class UniqueAssignmentListTest {
+
+    private final UniqueAssignmentList uniqueAssignmentList = new UniqueAssignmentList();
+
+    @Test
+    public void contains_nullAssignment_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueAssignmentList.contains(null));
+    }
+
+    @Test
+    public void contains_assignmentNotInList_returnsFalse() {
+        assertFalse(uniqueAssignmentList.contains(ALICE));
+    }
+
+    @Test
+    public void contains_assignmentInList_returnsTrue() {
+        uniqueAssignmentList.add(ALICE);
+        assertTrue(uniqueAssignmentList.contains(ALICE));
+    }
+
+    @Test
+    public void contains_assignmentWithSameIdentityFieldsInList_returnsTrue() {
+        uniqueAssignmentList.add(ALICE);
+        Assignment editedAlice = new AssignmentBuilder(ALICE).withAddress(VALID_ADDRESS_BOB).withTags(VALID_TAG_HUSBAND)
+                .build();
+        assertTrue(uniqueAssignmentList.contains(editedAlice));
+    }
+
+    @Test
+    public void add_nullAssignment_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueAssignmentList.add(null));
+    }
+
+    @Test
+    public void add_duplicateAssignment_throwsDuplicateAssignmentException() {
+        uniqueAssignmentList.add(ALICE);
+        assertThrows(DuplicateAssignmentException.class, () -> uniqueAssignmentList.add(ALICE));
+    }
+
+    @Test
+    public void setAssignment_nullTargetAssignment_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueAssignmentList.setAssignment(null, ALICE));
+    }
+
+    @Test
+    public void setAssignment_nullEditedAssignment_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueAssignmentList.setAssignment(ALICE, null));
+    }
+
+    @Test
+    public void setAssignment_targetAssignmentNotInList_throwsAssignmentNotFoundException() {
+        assertThrows(AssignmentNotFoundException.class, () -> uniqueAssignmentList.setAssignment(ALICE, ALICE));
+    }
+
+    @Test
+    public void setAssignment_editedAssignmentIsSameAssignment_success() {
+        uniqueAssignmentList.add(ALICE);
+        uniqueAssignmentList.setAssignment(ALICE, ALICE);
+        UniqueAssignmentList expectedUniqueAssignmentList = new UniqueAssignmentList();
+        expectedUniqueAssignmentList.add(ALICE);
+        assertEquals(expectedUniqueAssignmentList, uniqueAssignmentList);
+    }
```

Comment:
Nice test cases :) You may want to consider leaving a space between some parts of the code to make it more readable (e.g leaving a space between lines 79 and 80 and leaving another space between lines 81 and 82)